src/databricks/labs/remorph/reconcile/connectors/databricks.py

```diff
@@ -1,37 +1,49 @@
 import re
 
-from pyspark.errors import PySparkException
-from pyspark.sql import DataFrame
+from pyspark.sql import DataFrame, SparkSession
 from pyspark.sql.functions import col
+from sqlglot import Dialects
 
 from databricks.labs.remorph.reconcile.connectors.data_source import DataSource
+from databricks.labs.remorph.reconcile.connectors.secrets import SecretsMixin
 from databricks.labs.remorph.reconcile.recon_config import JdbcReaderOptions, Schema
-
-
-class DatabricksDataSource(DataSource):
-    def read_data(self, catalog: str, schema: str, query: str, options: JdbcReaderOptions) -> DataFrame:
+from databricks.sdk import WorkspaceClient
+
+
+class DatabricksDataSource(DataSource, SecretsMixin):
+
+    def __init__(
+        self,
+        engine: Dialects,
+        spark: SparkSession,
+        ws: WorkspaceClient,
+        secret_scope: str,
+    ):
+        self._engine = engine
+        self._spark = spark
+        self._ws = ws
+        self._secret_scope = secret_scope
+
+    def read_data(
+        self, catalog: str, schema: str, table: str, query: str, options: JdbcReaderOptions | None
+    ) -> DataFrame:
+        table_with_namespace = f"{catalog}.{schema}.{table}"
+        if catalog is None or catalog == "hive_metastore":
+            table_with_namespace = f"{schema}.{table}"
+        table_query = query.replace(":tbl", table_with_namespace)
         try:
-            table_query = self._get_table_or_query(catalog, schema, query)
-            df = self.spark.sql(table_query)
+            df = self._spark.sql(table_query)
             return df.select([col(column).alias(column.lower()) for column in df.columns])
-        except PySparkException as e:
-            error_msg = (
-                f"An error occurred while fetching Databricks Data using the following {table_query} in "
-                f"DatabricksDataSource : {e!s}"
-            )
-            raise PySparkException(error_msg) from e
+        except RuntimeError as e:
+            raise self._raise_runtime_exception(e, "data", table_query)
 
     def get_schema(self, catalog: str, schema: str, table: str) -> list[Schema]:
+        schema_query = self.get_schema_query(catalog, schema, table)
```

Comment:
staticmethod `get_schema_query` shouldn't be referred with `self`

Reply:
@ravit-db Can you address this.

Reply:
and also can make the method private

```suggestion
        schema_query = DatabricksDataSource._get_schema_query(catalog, schema, table)
```

Reply:
Same pattern applies everywhere